New entity target_doorstate.

lumayaa opened the pull request:
I got slightly annoyed by lmd_gate, so here we are.

- Added a WantPosition spawn key for func_door and lmd_door.
- target_doorstate validates referenced doors are in their desired positions before firing (keys door1-door6, values targetnames of doors)

keys:

-     {"target", "Fired when every referenced door matches its wantPosition."},
-     {"target2", "Fired when any referenced door is missing or does not match its wantPosition."},
-     {"door1", "Targetname of the first door to check."},
-     {"door2", "Targetname of the second door to check."},
-     {"door3", "Targetname of the third door to check."},
-     {"door4", "Targetname of the fourth door to check."},
-     {"door5", "Targetname of the fifth door to check."},
-     {"door6", "Targetname of the sixth door to check."},


Example:

/place target_doorstate * targetname,when_we_fire_this,target,to_fire_when_all_wantPositions_ok,target2,to_fire_when_wantPositions_arent_ok,door1,targetname_of_door1,door2,targetname_of_door2

lumayaa committed:
Init.

lumayaa committed:
Init.

lumayaa committed:
Merge branch 'DoorStateEntity' of https://github.com/lumayaa/Lugormod-v3 into DoorStateEntity

Jakendary commented:
This new entity will need to be added to the target ents documentation as part of this PR if possible so it is not forgotten about.

lumayaa committed:
Equip the new weapon if given a new one.

lumayaa committed:
New entity lmd_equalcheck.

This entity  compares a spawn key across up to six targetnames, firing EqualTarget when all values match and UnequalTarget otherwise.

lumayaa committed:
Documentation for lmd_equalcheck and target_doorstate.
